pkg/brownfield/listeners.go

```diff
@@ -49,7 +62,10 @@ func MergeListeners(listenerBuckets ...[]n.ApplicationGatewayHTTPListener) []n.A
 				Protocol: listener.Protocol,
 			}
 			if listener.HostName != nil {
-				listenerConfig.HostName = *listener.HostName
+				listenerConfig.setHostNames([]string{*listener.HostName})
+			}
+			if listener.Hostnames != nil && len(*listener.Hostnames) > 0 {
```

Comment:
the second setHostNames() always overrides the one for the HostName case above?

Reply:
only one of the above conditions can be true at a time.